addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/handler/NetworkHandler.java

```diff
@@ -75,46 +99,68 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     }
 
     @Override
-    public void newState(double state) {
-        State onlineState = state < 0 ? OnOffType.OFF : OnOffType.ON;
-        State timeState = new DecimalType(state);
-        updateState(CHANNEL_ONLINE, onlineState);
-        updateState(CHANNEL_TIME, timeState);
-    }
+    public void updatedDeviceState(double pingTime, PresenceDetectionType presenceDetectionType) {
+        // We do not notify the framework immediately if a device presence detection failed and
+        // the user configured retries to be > 1.
+        retryCounter = presenceDetectionType == PresenceDetectionType.NONE ? retryCounter + 1 : 0;
+        cache.setValue(pingTime);
+
+        if (retryCounter >= this.retries) {
+            updateState(CHANNEL_ONLINE, OnOffType.OFF);
+            updateState(CHANNEL_TIME, new DecimalType(0));
+            retryCounter = 0;
+        }
 
-    @Override
-    public void invalidConfig() {
-        updateStatus(ThingStatus.OFFLINE);
+        if (presenceDetectionType != PresenceDetectionType.NONE) {
+            lastSeen = new DateTimeType();
+            updateProperty(NetworkBindingConstants.PROPERTY_PRESENCE_DETECTION_TYPE, presenceDetectionType.name());
+            updateState(CHANNEL_ONLINE, OnOffType.ON);
+            updateState(CHANNEL_TIME, new DecimalType(pingTime));
+            updateState(CHANNEL_LASTSEEN, lastSeen);
+        }
     }
 
     int confValueToInt(Object value) {
         return value instanceof java.math.BigDecimal ? ((java.math.BigDecimal) value).intValue()
                 : Integer.valueOf((String) value);
     }
 
-    boolean confValueToBoolean(Object value) {
-        return value instanceof Boolean ? ((Boolean) value) : Boolean.valueOf((String) value);
+    @Override
+    public void dispose() {
+        networkService.stopAutomaticRefresh();
+        networkService = null;
     }
 
+    // Create a new network service and apply all configurations.
     @Override
     public void initialize() {
-        logger.debug("Initialize Network handler.");
-        this.networkService = new NetworkService();
+        this.networkService = new NetworkService(this);
         Configuration conf = this.getConfig();
 
         super.initialize();
 
-        networkService.setHostname(String.valueOf(conf.get(PARAMETER_HOSTNAME)));
+        try {
+            networkService.setHostname(String.valueOf(conf.get(PARAMETER_HOSTNAME)));
+        } catch (UnknownHostException e) {
+            logger.error("Configuration for hostname is faulty", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getLocalizedMessage());
```

Comment:
This is a good question. I would say it is a configuration error. It is actually an additional nice-to-have feature to allow people in the first place to use a dns name instead of an IP. The dns name should be at least resolvable in the startup process of the binding. Especially because the dns resolving does not talk to the target device but to a dns server.